packages/task/src/browser/provided-task-configurations.ts

```diff
@@ -34,11 +44,18 @@ export class ProvidedTaskConfigurations {
     @inject(TaskDefinitionRegistry)
     protected readonly taskDefinitionRegistry: TaskDefinitionRegistry;
 
+    readonly onStartUserInteractionEmitter: Emitter<TaskStartUserInteractionEvent> = new Emitter<TaskStartUserInteractionEvent>();
+
     private currentToken: number = 0;
     private nextToken = 1;
 
-    startUserAction(): number {
-        return this.nextToken++;
+    get onStartUserInteraction(): Event<TaskStartUserInteractionEvent> {
```

Comment:
minor: should we rename given the guidelines: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#event_names?

```suggestion
    get onDidStartUserInteraction(): Event<TaskStartUserInteractionEvent> {
```